test/arithmetic/floating_point/floating_point_sqrt_test.dart

```diff
@@ -0,0 +1,200 @@
+                '\t${expError.value} expected');
+      }
+    }
+  });
+
+  test('FP: square root with error flag high', () {
+    const exponentWidth = 8;
+    const mantissaWidth = 23;
+
+    final fv = FloatingPointValue(
+        sign: LogicValue.zero,
+        exponent: LogicValue.filled(exponentWidth, LogicValue.one),
+        mantissa: LogicValue.filled(mantissaWidth, LogicValue.zero));
+
+    final fp = FloatingPoint(
+        exponentWidth: exponentWidth, mantissaWidth: mantissaWidth);
+
+    for (final sqrtDUT in [
+      FloatingPointSqrtSimple(fp),
+    ]) {
+      final testCases = [
+        fv.clonePopulator().positiveInfinity.negate(),
+        fv.clonePopulator().one.negate(),
+      ];
+
+      for (final test in testCases) {
+        final fv = test;
+        final expError = Const(1);
+
+        fp.put(fv);
+        final eOut = sqrtDUT.error;
+
+        expect(eOut.value, equals(expError.value),
+            reason: 'error =\n'
+                '\t${eOut.value} actual\n'
+                '\t${expError.value} expected');
+      }
+    }
+  });
+
+  test('FP: targeted normalized sqrt', () {
+    const exponentWidth = 8;
+    const mantissaWidth = 23;
+
+    final fp = FloatingPoint(
+        exponentWidth: exponentWidth, mantissaWidth: mantissaWidth);
+
+    for (final sqrtDUT in [
+      FloatingPointSqrtSimple(fp),
+    ]) {
+      final testCases = [
+        144.0,
+        288.0,
+        3.567,
+        1123.5,
+        17.0,
+        92.5,
+        100.0,
+        85.672,
+      ];
+
+      for (final test in testCases) {
+        final fv = FloatingPointValue.populator(
+                exponentWidth: exponentWidth, mantissaWidth: mantissaWidth)
+            .ofDouble(test);
+
+        fp.put(fv);
+
+        final compResult = sqrtDUT.sqrtR;
+        final compError = sqrtDUT.error;
+
+        final expResult = FloatingPointValue.populator(
+                exponentWidth: exponentWidth, mantissaWidth: mantissaWidth)
+            .ofDouble(sqrt(test));
+        final expError = Const(0);
+        expect(compResult.floatingPointValue.toDouble(),
+            equals(expResult.toDouble()),
+            reason: '\t${fp.floatingPointValue} '
+                '(${fp.floatingPointValue.toDouble()}) =\n'
+                '\t${compResult.floatingPointValue}'
+                '(${compResult.floatingPointValue.toDouble()}) actual\n'
+                '\t$expResult (${expResult.toDouble()}) expected');
+
+        expect(compError.value, equals(expError.value),
+            reason: 'error =\n'
+                '\t${expError.value} actual\n'
+                '\t${expError.value} expected');
+      }
+    }
+  });
+
+  test('FP: random number sqrt', () {
+    const exponentWidth = 5;
+    const mantissaWidth = 9;
+
+    final fp = FloatingPoint(
+        exponentWidth: exponentWidth, mantissaWidth: mantissaWidth);
+
+    final sqrtDUT = FloatingPointSqrtSimple(fp);
+    final rand = Random(513);
+    for (var i = 0; i < 4096; i++) {
```

Comment:
Can you compute the limit (4096) from eWidth and mWidth?

Reply:
Possibly? Not 100% the question here. I wanted to choose something reasonably large since we do end up discarding negative numbers

Reply:
Thinking ahead to a variable width component for narrow and wider sqrts -- it would be great if the testing framework scaled.

Reply:
Agreed, removed magic numbers :)